docs/rules/no-use-before-define.md

```diff
@@ -143,4 +143,10 @@ function baz() {
 }
 
 var foo = 1;
+
+
+const a = () => b(); // ok
+
+const b = () => {};
+
```

Comment:
This looks good, just minor corrections: 

* It's ok to have `// ok` comments in examples, but since this particular document doesn't have that in other examples, maybe we should remove it?
* It would be nice to remove blank lines 144 and 149, and thus group lines of code that represent a single example.

Also, can we add some other examples in the same `### variables` section:

* An incorrect example with an arrow function.
* An incorrect and correct example with function expressions.

Reply:
> An incorrect example with an arrow function.

and incorrect example with arrow function like this  ?
Like this , as a single example to show the usage better

_thie snippet covering max cases of variables options with function_

-  Incorrect

```js
b()
const b = () => {};
```
- Correct

```js
const a = () => b();

function c(){ return b() } 

const d = function(){ return b() }

const b = () => {};

```

> An incorrect and correct example with function expressions.

- Incorrect

```js
function a(){ return b() }; 

const c = () => b()

function b(){ }
```

- Correct 

```js

function a() {}

function b(){ return a() }

```

@mdjermanovic  What do you think ?

Reply:
> _thie snippet covering max cases of variables options with function_
> 
> * Incorrect
> 
> ```js
> b()
> const b = () => {};
> ```
> 
> * Correct
> 
> ```js
> const a = () => b();
> 
> function c(){ return b() } 
> 
> const d = function(){ return b() }
> 
> const b = () => {};
> ```

Looks great! Maybe we could just rename variables to be a bit more readable:

```js
f()
const f = () => {};
```

```js
const a = () => f();
function b() { return f(); } 
const d = function() { return f(); }
const f = () => {};
```

Reply:
> > An incorrect and correct example with function expressions.
> 
> * Incorrect
> 
> ```js
> function a(){ return b() }; 
> 
> const c = () => b()
> 
> function b(){ }
> ```
> 
> * Correct
> 
> ```js
> function a() {}
> 
> function b(){ return a() }
> ```

As for function expression examples, the variable being referenced should be initialized to a function expression (similar to examples with an arrow function expression).

Reply:
like this  ?

```js
const a = function(){}
function b(){ return a() }
```

> ```js
> function a() {}
> 
> function b(){ return a() }
> ```

This is also lint free, should  I keep both ?

Reply:
> like this ?
> 
> ```js
> const a = function(){}
> function b(){ return a() }
> ```

Something like this:

```js
const a = function() { return b(); }
const b = function() {}
```

because it's an example where `{ "variables": false }` changes the behavior of this rule (if it wasn't set, `return b();` would be error).

> ?
> 
> > ```js
> > function a() {}
> > 
> > function b(){ return a() }
> > ```
> 
> This is also lint free, should I keep both ?

Better not, since this isn't related to `"variables"` option so it might be confusing.

Reply:
@mdjermanovic would it make sense to add a note about how function declaration and variable declaration distinct for `function`

like this 

```js
const a  = function(){  } // VariableDeclaration : variables options works for this
function a(){  }  // FunctionDeclaration  : variables options doesnt works for this
```

?

Reply:
I think it's a good idea!

Since there are already examples, maybe we could just add the following text to the end of the `### functions` section:

> This option allows references to function declarations. For function expressions and arrow functions, please see the `"variables"` option.